### PR TITLE
Support for Kubernetes Audit webhook

### PR DESCRIPTION
**Pull request by gianarb**
This PR adds a new webhook to catch Kubernetes audit logs.
There is a README.md with more information about how to try it and how
Kubernetes Audit works.

TODO:
- [x] Verify if the generated point has sense /cc @goller 
- [ ] I still need to add few more fields and tags inside the point

Signed-off-by: Gianluca Arbezzano <gianarb92@gmail.com>

### Required for all PRs:

- [x] Signed [CLA](https://influxdata.com/community/cla/).
- [x] Associated README.md updated.
- [x] Has appropriate unit tests.

### CODE REVIEW

File: plugins/inputs/webhooks/kubernetes_audit/audit_webhooks.go [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+	if err != nil {
+		http.Error(w, string(b), http.StatusInternalServerError)
+		return
+	}
+	for ii := 0; ii < len(audits.Items); ii++ {
+		tags := map[string]string{}
+		fields := map[string]interface{}{}
+		audit := audits.Items[ii]
+		tags["audit_id"] = string(audit.AuditID)
+		tags["level"] = string(audit.AuditID)
```
Comment (goller) [inline plugins/inputs/webhooks/kubernetes_audit/audit_webhooks.go]: I think this is supposed to be Level

File: plugins/inputs/webhooks/kubernetes_audit/audit_webhooks.go [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+		tags["verb"] = string(audit.Verb)
+		tags["user_username"] = audit.User.Username
+		tags["user_uid"] = audit.User.UID
+		tags["resource_name"] = audit.ObjectRef.Name
+		tags["namespace"] = audit.ObjectRef.Namespace
+		tags["resource_type"] = audit.ObjectRef.Resource
+		tags["resource_version"] = audit.ObjectRef.ResourceVersion
+
+		//"groups": ["system:authenticated"]
+		//audit.User.Groups
```
Comment (gianarb) [inline plugins/inputs/webhooks/kubernetes_audit/audit_webhooks.go]: @danielnelson Hello! do you have some feedback about how we should manage this kind of data?

File: plugins/inputs/webhooks/kubernetes_audit/audit_webhooks.go [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+		// "sourceIPs": ["192.168.99.100"],
+		//audit.SourceIPs
```
Comment (gianarb) [inline plugins/inputs/webhooks/kubernetes_audit/audit_webhooks.go]: @danielnelson Hello! do you have some feedback about how we should manage this kind of data?

File: plugins/inputs/webhooks/kubernetes_audit/audit_webhooks.go [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+		}
+
+		//"groups": ["system:authenticated"]
+		//audit.User.Groups
+
+		// "sourceIPs": ["192.168.99.100"],
+		//audit.SourceIPs
+
+		fields["request_uri"] = string(audit.RequestURI)
+		k.acc.AddFields("kubernetes_audit", fields, tags, audit.RequestReceivedTimestamp.Time)
```
Comment (danielnelson) [inline plugins/inputs/webhooks/kubernetes_audit/audit_webhooks.go]: This schema is fine but take a little time to consider changing this from a logging style plugin to a plugin which collects metrics.  Think about what sort of metrics you will like to graph and alert on.

**Comment by gianarb** [comment]
It is running in our system since 3weeks now. We will keep running it.
Thanks for all the feedback, I am gonna address them

**Comment by russorat** [comment]
This is looking for a new owner. If a community member would like to take it across the finish line, then they should take it over.

**Comment by sjwang90** [comment]
Closing this PR due to inactivity. If someone is interested on taking this work back up, please feel free to re-open the PR and take over the code.

**Pull request closed without merging**